internal/services/compute/virtual_machine_implicit_data_disk_resource.go

```diff
@@ -0,0 +1,462 @@
+package compute
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2022-03-02/snapshots"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2024-03-01/virtualmachines"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.Resource                   = VirtualMachineImplicitDataDiskResource{}
+	_ sdk.ResourceWithUpdate         = VirtualMachineImplicitDataDiskResource{}
+	_ sdk.ResourceWithCustomImporter = VirtualMachineImplicitDataDiskResource{}
+)
+
+type VirtualMachineImplicitDataDiskResource struct{}
+
+func (r VirtualMachineImplicitDataDiskResource) ModelObject() interface{} {
+	return &VirtualMachineImplicitDataDiskResourceModel{}
+}
+
+type VirtualMachineImplicitDataDiskResourceModel struct {
+	Name                    string `tfschema:"name"`
+	VirtualMachineId        string `tfschema:"virtual_machine_id"`
+	Lun                     int64  `tfschema:"lun"`
+	Caching                 string `tfschema:"caching"`
+	CreateOption            string `tfschema:"create_option"`
+	DeleteOption            string `tfschema:"delete_option"`
+	DiskSizeGb              int64  `tfschema:"disk_size_gb"`
+	SourceResourceId        string `tfschema:"source_resource_id"`
+	WriteAcceleratorEnabled bool   `tfschema:"write_accelerator_enabled"`
+}
+
+func (r VirtualMachineImplicitDataDiskResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.DataDiskID
+}
+
+func (r VirtualMachineImplicitDataDiskResource) ResourceType() string {
+	return "azurerm_virtual_machine_implicit_data_disk"
+}
+
+func (r VirtualMachineImplicitDataDiskResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"virtual_machine_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateVirtualMachineID,
+		},
+
+		"lun": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntAtLeast(0),
+		},
+
+		"caching": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(virtualmachines.CachingTypesNone),
```

Comment:
See the `None` topic in the Contributor Guide

Reply:
code updated. Thanks.